scripts/contacts/migrateContactsV1toV2.js

```diff
@@ -94,50 +99,50 @@ const migrateContactV1toV2 = (contact, accountId, contactAccountId) => {
   }
 }
 
-const createContactAccount = (account, accountName) => ({
+const createContactAccount = accountName => ({
   canLinkContacts: true,
   shouldSyncOrphan: true,
   lastSync: null,
   lastLocalSync: null,
   name: accountName,
   type: KONNECTOR_APP_NAME,
-  sourceAccount: account._id,
+  sourceAccount: null,
   version: ACCOUNT_APP_VERSION
 })
 
-const getAccountName = contacts => {
-  const contactWithFrom = contacts.find(contact =>
-    has(contact, 'metadata.google.from')
-  )
-  return contactWithFrom ? contactWithFrom.metadata.google.from : ''
-}
-
 async function doMigrations(dryRun, api, logWithInstance) {
   const contacts = await api.fetchAll(DOCTYPE_CONTACTS)
   if (contacts.length === 0) {
     logWithInstance('No contacts, nothing to migrate')
     return
   }
 
-  const accounts = await api.fetchAll(DOCTYPE_ACCOUNTS)
-  const konnectorAccount = accounts.find(doc => doc.account_type === 'google')
-  const accountName = getAccountName(contacts)
-
-  let accountId, contactAccountId
-  if (dryRun) {
-    accountId = 'dryRunAccountId'
-    contactAccountId = 'dryRunContactAccountId'
-    const contactAccount = createContactAccount({ _id: accountId }, accountName)
-    logWithInstance('Dry run: would create contact account', contactAccount)
-  } else if (konnectorAccount) {
-    const contactAccount = createContactAccount(konnectorAccount, accountName)
-    const response = await api.update(DOCTYPE_CONTACTS_ACCOUNT, contactAccount)
-    accountId = konnectorAccount._id
-    contactAccountId = response.id
-  }
+  const accountsNames = uniq(
+    contacts
+      .map(contact => get(contact, 'metadata.google.from'))
+      .filter(name => name !== undefined)
+  )
+
+  let contactsAccountsMapping = {}
+  await Promise.all(
+    accountsNames.map(async name => {
+      const contactAccount = createContactAccount(name)
+      if (dryRun) {
+        logWithInstance('Dry run: would create contact account', contactAccount)
+        contactsAccountsMapping[name] = `contactAccount-${name}`
+      } else {
+        const response = await api.update(
+          DOCTYPE_CONTACTS_ACCOUNT,
+          contactAccount
+        )
```

Comment:
A query in a map. Best way to ddos the db. But if I understand correctly, this is only here for the accounts and we don't expect accountsName.length to be very big, right?

Reply:
according to clochix's list its 10 max and mostly 1 or 2. But good catch, I will try to change it